apps/api/src/app/messages/e2e/remove-messages-by-transactionId.e2e.ts

```diff
@@ -0,0 +1,137 @@
+      `${session.serverUrl}/v1/subscribers`,
+      {
+        subscriberId: '123456',
+        name: 'broadcast subscriber',
+      },
+      {
+        headers: {
+          authorization: `ApiKey ${session.apiKey}`,
+        },
+      }
+    );
+
+    const response = await axiosInstance.post(
+      `${session.serverUrl}/v1/events/trigger/broadcast`,
+      {
+        name: template.triggers[0].identifier,
+        payload: {
+          email: 'new-test-email@gmail.com',
+          firstName: 'Testing of User Name',
+          urlVar: '/test/url/path',
+        },
+      },
+      {
+        headers: {
+          authorization: `ApiKey ${session.apiKey}`,
+        },
+      }
+    );
+
+    await session.awaitRunningJobs(template._id);
+
+    const transactionId = response.data.data.transactionId;
+
+    const messages = await messageRepository.find({
+      _environmentId: session.environment._id,
+      _organizationId: session.organization._id,
+      transactionId: transactionId,
+    });
+
+    expect(messages.length).to.be.greaterThan(0);
+
+    await axiosInstance.delete(`${session.serverUrl}/v1/messages/transaction/${transactionId}`, {
+      headers: {
+        authorization: `ApiKey ${session.apiKey}`,
+      },
+    });
+
+    const result = await messageRepository.find({
+      transactionId: transactionId,
+      _environmentId: session.environment._id,
+      _organizationId: session.organization._id,
+    });
+
+    expect(result.length).to.equal(0);
+  });
+
+  it('should delete messages by transactionId and channel', async function () {
+    const response = await axiosInstance.post(
+      `${session.serverUrl}/v1/events/trigger/broadcast`,
+      {
+        name: template.triggers[0].identifier,
+        payload: {
+          email: 'new-test-email@gmail.com',
+          firstName: 'Testing of User Name',
+          urlVar: '/test/url/path',
+        },
+      },
+      {
+        headers: {
+          authorization: `ApiKey ${session.apiKey}`,
+        },
+      }
+    );
+
+    await session.awaitRunningJobs(template._id);
+    const transactionId = response.data.data.transactionId;
+
+    const messages = await messageRepository.find({
+      _environmentId: session.environment._id,
+      _organizationId: session.organization._id,
+      transactionId: transactionId,
+      channel: ChannelTypeEnum.EMAIL,
+    });
+
+    expect(messages.length).to.be.greaterThan(0);
+
+    await axiosInstance.delete(
+      `${session.serverUrl}/v1/messages/transaction/` + transactionId + '?channel=' + ChannelTypeEnum.EMAIL,
+      {
+        headers: {
+          authorization: `ApiKey ${session.apiKey}`,
+        },
+      }
+    );
+
+    const result = await messageRepository.find({
+      transactionId: transactionId,
+      _environmentId: session.environment._id,
+      _organizationId: session.organization._id,
+      channel: ChannelTypeEnum.EMAIL,
+    });
+
+    expect(result.length).to.equal(0);
```

Comment:
could you please update this test with a scenario:
- call a trigger for template with in-app and email
- check that in-app and email messages were sent
- delete by transaction and email channel
- check that the only email messages are deleted but in-app still present

Thank you! 🙌